int16 pin
int16 command